### ISSUE TITLE
Add a cron monitore

### ISSUE THREAD

**Issue by Dryusdan** [louislam/uptime-kuma#1482]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Monitor, UI Feature

### 🔖 Feature description

To monitore cron, it's easier to send OK/NOK with curl (like healthcheck.io) or if cron don't execute after specific time it raise an alert.

I see push monitore and I think it can have a little improvement : set Heartbeat interval in second or cron execution format.

For exemple I want to monitore a cron who execute each monday at 3AM, I can define 604800 in interval, or define `0 3 * * 1` in interface to monitore the cron

### ✔️ Solution

Add in push monitor a way to select interval or cron execution time format.

### ❓ Alternatives

Healtcheck.io can work (it work) but multiplying the tools can be a bit of a headache

**Comment by mathiskir** [comment]
I don't think this will be a thing for uptime kuma. Its more for web things like websites, dns records and so one but not for cronjobs.

**Comment by Eidenz** [comment]
I think this is a great idea. I don't see why uptime-kuma couldn't be used to monitor cronjobs, it would do the job just fine.

If there are push monitors, why not time range push.

**Comment by Seele0oO** [comment]
That's what I think too. The time interval alone is not sufficient for monitoring the status of the task.

**Comment by HujinoKun** [comment]
It would be great to have an api that would allow to send POSTs to know if our cronjob was successful or not.

A bit like betteruptime does

**Comment by Haelle** [comment]
I think it's a good feature for Kuma, web services often have cron jobs hidden they rely on.

**Comment by hypery2k** [comment]
I'm also interested in this functionality, similiar to what cronitor does: https://cronitor.io/cron-job-monitoring

I the feature itself would be interesting, I'm willing to create a PR for that

**Comment by CommanderStorm** [comment]
Changing the monitor sheduler is a lot of work.
Current plan is here if you want to contribute to this PR (via targeting the `queue` instead of the `master` branch):
- https://github.com/louislam/uptime-kuma/pull/3447

**Comment by hausen1012** [comment]
I also need this feature, but it seems like this issue has been open for quite a while.there are any technical implementation issues?

**Comment by CommanderStorm** [comment]
> Changing the monitor sheduler is a lot of work.

It is a lot of work and touches some of the core of what we do.
Also fairly risky..
